botocore/session.py

```diff
@@ -796,6 +799,37 @@ def create_client(self, service_name, region_name=None, api_version=None,
             client_config=config, api_version=api_version)
         return client
 
+    def get_available_partitions(self):
+        """Lists the available partitions found on disk
+
+        :rtype: list
+        :return: Returns a list of partition names (e.g., ["aws", "aws-cn"])
+        """
+        resolver = self.get_component('endpoint_resolver')
+        return resolver.get_available_partitions()
+
+    def get_available_regions(self, service_name, partition_name='aws',
```

Comment:
Sounds good, it'd be worth updating the docstring to say that endpointprefix and service name are both supported.